core/root/config/templates/cluster.yaml

```diff
@@ -1555,3 +1555,7 @@ hostOS:
 #    kubernetes-colour: light-blue
 #    kube-aws-colour: light-blue
 
+# disableOOMKiller - [experimental] disable the linux kernel "Out Of Memory" process killer
```

Comment:
Makes sense. So, you're saying that this option, when enabled, could make processes like dockerd and kubelet that highly depends on overcommit unstable, due to that they are now OOM killed at the time of mem request, rather than actual memory shortage?

Can we make a heuristic to choose which process are generally "vital system components" ? In that case, would it be possible to for us to alternatively set `oom_adj` for those processes rather than changing `overcommit_memory` and `overcommit_ratio`?

Reply:
best way to make apps stable is to ensure that app doesn't try to allocate beyond what limits says. In case of JVM that is roughly equivalent to setting `-Xmx` to limit size - 256 MB (or better -512).

Reply:
As a platform engineer, I don't have a lot of levers to ensure all apps are stable - and so I'm interested in ways in which we can protect the platform (whilst we and PRET like teams) encourage better application quality.

@mumoshu I think it is theoretically possible to identify critical platform processes and mark them so that they can't be OOM killed but it will be hard because we will have to find a way of finding their PIDs and have some sort of service responsible for updating the oom_score_adj:
`echo -1000 > /proc/$PID/oom_score_adj`

Reply:
> ways in which we can protect the platform 

set resource limit not so far from  for resource requests, allowing no more than 20-50% overcommit, then it will be individual container (cgroup) which hits it's limit and therefore will be OOMkilled, without affecting platform or other apps.

If you want to have 100% guarantees, then lower limit <> requests gap to zero, forcing requests to be == limits, then you achieve same platform protection , without giving up memory overcommits within each container.

Reply:
Yes, I do see what you are saying but that assumes that I have some way of exerting control over the resource and limits that individual dev teams are setting in their charts.  I like the look of the [Vertical Pod Auto-Scaler](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler) which will possibly be a way of curbing application owners omissions or excesses but it is very new and quite crude - and more importantly, opt-in.  We wouldn't experience OOM if the requests = limits and were set correctly.